addons/account/models/account_account.py

```diff
@@ -148,35 +151,37 @@ def _field_to_sql(self, alias: str, fname: str, query: (Query | None) = None, fl
             return self.with_company(self.env.company.root_id).sudo()._field_to_sql(alias, 'code_store', query, flush)
         if fname == 'placeholder_code':
             query.add_join(
-                'JOIN',
+                'LEFT JOIN',
                 'account_first_company',
                 SQL(
                     """(
-                        SELECT DISTINCT ON (rel.account_account_id)
-                               rel.account_account_id AS account_id,
-                               rel.res_company_id AS company_id,
-                               SPLIT_PART(res_company.parent_path, '/', 1) AS root_company_id,
-                               res_company.name AS company_name
-                          FROM account_account_res_company_rel rel
-                          JOIN res_company
-                            ON res_company.id = rel.res_company_id
-                         WHERE rel.res_company_id IN %(authorized_company_ids)s
-                      ORDER BY rel.account_account_id, company_id
+                        SELECT DISTINCT ON (account_companies.account_account_id)
+                               account_companies.account_account_id AS account_id,
+                               account_companies.res_company_id AS company_id,
```

Comment:
The order by clause probably need a account_companies.res_company_id because of the distinct on, to be deterministic.
I've no deep knowledge of this business part, so maybe @qdp-odoo have a better suggestion.

Reply:
I think it's not needed since we are doing
`JOIN account_companies ON account_companies.res_company_id = active_company_parent_ordering.id`.
So that JOIN will not increase the number of rows and since we are ordering by `active_company_parent_ordering.idx` the order will be entirely determined.